### ISSUE TITLE
Regression with `--verifytypes` for imported TypeVars

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#4600]
**Describe the bug**

Prior to `v1.1.293`, running `pyright --verifytypes $pkg --ignoreexternal` didn't report any type errors for the following code + file structure:

```
foo/
  __init__.py
  bar.py
  baz.py
```

```py
# bar.py
from typing import Type

ModelT = TypeVar('ModelT')
```

```py
# baz.py
from typing import Type
from .bar import ModelT

def foo(model: Type[ModelT]) -> ModelT:
    ...
```

Pyright reports the following error:

```
foo.baz.foo
  baz.py:4:5 - error: Type of parameter "model" is partially unknown
    Parameter type is "Type[ModelT@foo]"
  baz.py:4:5 - error: Return type is partially unknown
    Return type is "ModelT@foo"
```

**Expected behavior**

It should not be reported as partially unknown.

**VS Code extension or command-line**

Pyright CLI `v1.1.293`.

**Additional context**

Defining the TypeVar in the same module as it is used fixes the error, e.g.

```py
from typing import Type, TypeVar

ModelT = TypeVar('ModelT')

def foo(model: Type[ModelT]) -> ModelT:
    ...
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published.

**Comment by RobertCraigie** [comment]
Thank you!
